core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1349,6 +1339,29 @@ class DAGScheduler(
               s"longer running")
           }
 
+          if (mapStage.rdd.isBarrier()) {
```

Comment:
this won't handle the case when the barrier stage is two shuffle stages back, right?  you might need to go back many stages when there is a fetch failure.

Reply:
btw what I really mean here, is that I think you actually need to do this every time to create task set.  Check if the task set is for a barrier stage, and if so, then launch all the tasks, regardless of existing output.

Reply:
Previously I was thinking that by unregistering shuffle output, we may avoid modify the submit missing tasks logic. Now I realized you have to launch all the tasks for taskSet of a barrier stage anyway, so maybe the approach you mentioned is cleaner, I'll try to submit a follow up PR on that. Thanks!

Reply:
hi @jiangxb1987 was this addressed?  sorry if I missed it